gremlin-client/src/connection.rs

```diff
@@ -13,30 +20,49 @@ impl std::fmt::Debug for ConnectionStream {
 impl ConnectionStream {
     fn connect(options: ConnectionOptions) -> GremlinResult<Self> {
         let connector = match options.tls_options.as_ref() {
-            Some(option) => Some(
+            Some(option) => Some(Connector::NativeTls(
                 option
                     .tls_connector()
                     .map_err(|e| GremlinError::Generic(e.to_string()))?,
-            ),
+            )),
             _ => None,
         };
 
-        let client = ClientBuilder::new(&options.websocket_url())
-            .map_err(|e| GremlinError::Generic(e.to_string()))?
-            .connect(connector)?;
+        let request = options
+            .websocket_url()
+            .into_client_request()
+            .map_err(|e| GremlinError::Generic(e.to_string()))?;
+        let uri = request.uri();
+        let mode = uri_mode(uri).map_err(|e| GremlinError::Generic(e.to_string()))?;
+        let host = request
+            .uri()
+            .host()
+            .ok_or_else(|| GremlinError::Generic("No Hostname".into()))?;
+        let port = uri.port_u16().unwrap_or(match mode {
+            Mode::Plain => 80,
+            Mode::Tls => 443,
+        });
+        let mut stream = TcpStream::connect((host, port))
+            .map_err(|e| GremlinError::Generic(format!("Unable to connect {e:?}")))?;
+        NoDelay::set_nodelay(&mut stream, true)
+            .map_err(|e| GremlinError::Generic(e.to_string()))?;
```

Comment:
At the risk of embarrassing myself this was largely inspired from https://github.com/snapview/tungstenite-rs/blob/4729b0f00006c6f73a17c9248ae4e6005ec88d30/src/client.rs#L51-L59. If there's a preferred why you'd like this stream created, happy to change